docs/development/workflow/maintainer_workflow.rst

```diff
@@ -7,105 +7,109 @@ Workflow for Maintainers
-    git fetch upstream-rw --tags
+However, this feature should *not* be used in the following cases:
 
-Let's say you have some changes that need to go into trunk
-(``upstream-rw/main``).
+* You should not enable auto-merge on your own pull request unless
+  other maintainers are okay with it. There might be exceptions but
+  they should be handled on a case-by-case basis.
+* If a pull request requires job(s) that is not required to pass
+  branch protection rules (e.g., allowed to fail job, cron jobs) to pass
+  to be acceptable (say, it touches remote data or has very big changes that need extra checks).
+* If a pull request requires approvals from multiple reviewers and
+  not all the reviews are in but you are happy with it.
+* If you are unsure of all the implications of using this feature.
 
-The changes are in some branch that you are currently on. For example, you are
-looking at someone's changes like this::
+.. _maintainer-pr-takeover:
 
-    git remote add someone https://github.com/someone/astropy.git
-    git fetch someone
-    git branch cool-feature --track someone/cool-feature
-    git checkout cool-feature
+==========================
+Taking over a pull request
+==========================
 
-So now you are on the branch with the changes to be incorporated upstream. The
-rest of this section assumes you are on this branch.
-
-A few commits
--------------
-
-If there are only a few commits, consider rebasing to upstream::
-
-    # Fetch upstream changes
-    git fetch upstream-rw
+Remember that, if you take over a pull request, you will show up as
+co-author on the GitHub commit history. Furthermore, the author will
+need to know how to grab your changes back onto their local branch,
+which might be daunting for new contributors. If you are not sure
+the author would be okay with this, ask first.
 
-    # Rebase
-    git rebase upstream-rw/main
+If author is not okay with this or has disabled "edit by maintainers"
+option, you can still cherry-pick their commits onto a new branch of
+your own and create a new pull request off that, which would close out
+the original pull request.
 
-Remember that, if you do a rebase, and push that, you'll have to close any
-github pull requests manually, because github will not be able to detect the
-changes have already been merged.
+But let's say the author is okay with you taking over...
 
-A long series of commits
-------------------------
+First, check out the ``astropy`` repository and :ref:`set_upstream_main`.
+Now, you need to point a remote to the pull request author's fork.
+In this example, the author's username is ``octocat`` and the pull request
+branch name is ``cool-feature`` that is tied to pull request number 99999::
 
-If there are a longer series of related commits, consider a merge instead::
+    git remote add octocat git@github.com:octocat/astropy.git
+    git fetch octocat cool-feature
+    git checkout octocat/cool-feature -b pr99999
```

Comment:
I prefer using [`git switch`](https://git-scm.com/docs/git-switch) because I have to type less before I can use tab-completion.
If there is only one remote with a branch named `cool-feature` then the following will work:
```suggestion
    git switch cool-feature
```
But if there are several remotes with that branch then the remote can be specified explicitly:
```suggestion
    git switch --track octocat/cool-feature
```
And the following allows naming the local branch differently:
```suggestion
   git switch -c pr99999 --track octocat/cool-feature
```

Reply:
Good to know. I'll take the third suggestion. Thanks!